src/app/scripts/cac/pages/cac-pages-home.js

```diff
@@ -343,19 +321,41 @@ CAC.Pages.Home = (function ($, ModeOptions,  MapControl, TripOptions, SearchPara
     }
 
     /**
-     * When user clicks the Directions link on a destination, send them to the directions tab
-     * with that destination (whether or not there's an origin set)
+     * When user clicks the Directions link on a place card
      */
     function clickedDestinationDirections(event) {
         event.preventDefault();
 
         var placeCard = $(event.target).closest(options.selectors.placeCard);
         var placeId = placeCard.data('destination-id');
-        UserPreferences.setPreference('placeId', placeId);
+
         var destination = {
+            id: placeId,
             address: placeCard.find(options.selectors.placeCardName).text(),
             location: { x: placeCard.data('destination-x'), y: placeCard.data('destination-y') }
         };
+
+        goToDestinationDirections(placeId, destination);
+    }
+
+    /**
+     * When user clicks 'Get Directions' in a destination marker popup
+     */
+    function clickedDestinationDirectionsPopup(event, place) {
+        // match the format expected
+        place.location = {
+            x: place.point.coordinates[0],
+            y: place.point.coordinates[1]
+        };
+        goToDestinationDirections(place);
```

Comment:
Not sure if I'm missing something, but that's not the behavior I'm seeing.

Clicking 'Get Directions' in the popup (as for this handler), clicking on the place card, and autocomplete all fill in the place name and not the address in the typeahead form box.